transcom/mymove
# [MB-15939] Update GHC API to Retrieve Documents Stored by Prime for Service Items

## [Jira ticket](https://dp3.atlassian.net/browse/MB-15939?atlOrigin=eyJpIjoiYjc0MDVmZjk1YTk0NDk2MjlhY2MwMTFlOGU1YjI0OGMiLCJwIjoiaiJ9)

## Summary
-Updated `move_task_order_fetcher` and `mto_service_items.go` to include `ServiceRequestDocumentUploads`
-Added tests for `mto_service_items_test.go` and `move_task_order_fetcher.go`
-Created `ServiceRequestDocument.yaml` and `ServiceRequestDocuments.yaml`
-Updated `MTOServiceItem.yaml`

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Log into office app as prime simulator
2. Select a move. Scroll down to "Service Items" and upload some documents.
3. Log in as TOO and find the same move in the queue. 
4. Open DevTools, click on "Network" tab, scroll to "mto_service_items", open "preview" on the right side. Select a service item. You should see `serviceRequestDocuments`.

Run make server_test.

### Backend


## Screenshots

<img width="1167" alt="Screenshot 2023-06-21 at 12 36 10 PM" src="https://github.com/transcom/mymove/assets/110994428/827a3cd2-ffcf-48e2-ba81-2084cd6525ac">
